src/python/pants/help/help_info_extracter.py

```diff
@@ -175,7 +175,7 @@ def get_option_help_info(self, args, kwargs):
       deprecated_message = 'DEPRECATED. {} removed in version: {}'.format(deprecated_tense,
                                                                           removal_version)
     removal_hint = kwargs.get('removal_hint')
-    choices = ', '.join(str(kwargs.get('choices'))) if kwargs.get('choices') else None
+    choices = ', '.join(str(choice) for choice in kwargs.get('choices')) if kwargs.get('choices') else None
```

Comment:
If you want to avoid evaluating `kwargs.get('choices')` twice you can do something like: 
```
', '.join(str(choice) for choice in kwargs.get('choices', [])) or None
```

Reply:
If kwargs.get('choices') would have returned None, the `get` with default will return `[]`, the join will return an empty string, and the `or None` will return `None`.